### ISSUE TITLE
sampler test cases

### ISSUE THREAD

**Issue by raphaelhoffmann** [HazyResearch/deepdive#284]
Currently, the test cases for the sampler use binary files as input which makes it difficult or impossible to understand and edit test cases. It would be great if we had a compact, readable, and self-contained text representation (eg. Alchemy format with atoms, predicates, weights). 

Across different test cases, we then want to keep all parameters fixed except for the issue that we are testing. Eg. the epsilon from target expectation, number of samples, etc may be fixed, while we vary factors.

**Comment by zhangce** [comment]
@netj draws a similar point in https://github.com/HazyResearch/sampler/commit/9296be214f40acda711d84ff4045154deccc5f2d#commitcomment-11008261 that we could easily use the "format-converter in DeepDive" to produce the binary given a text input. I think this should be pretty easy to do (maybe also with the Bats thing for testing that @netj used in that ticket?)

Ce

**Comment by raphaelhoffmann** [comment]
That sounds great, that solves the issue!

Longer term, we may still want to switch to an even more readable representation. I think the text format is still pretty hard to parse for a human, especially when compared to the Alchemy format, but I think for now it's great.

Another minor point: I like using a testing framework that is less complex than C++, but we are now using 4 different testing frameworks (for C++, scala, python, bash). It seems we could simplify by sticking to python for this. I imagine a future version of deepdive which only uses C++ and python.
